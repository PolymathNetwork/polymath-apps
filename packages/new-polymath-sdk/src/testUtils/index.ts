import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
