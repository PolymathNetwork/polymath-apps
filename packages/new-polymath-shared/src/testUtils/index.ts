export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
