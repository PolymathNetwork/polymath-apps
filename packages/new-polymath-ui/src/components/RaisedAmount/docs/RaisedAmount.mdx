---
name: RaisedAmount
menu: Components
---

import { PropsTable, Playground } from 'docz';

import { RaisedAmount } from '../';

# RaisedAmount

<PropsTable of={RaisedAmount} />

## Basic usage

<Playground>
  <RaisedAmount
    title="Amount Of Funds the STO Will Raise"
    primaryAmount={10000}
    tokenAmount={10000}
    tokenUnit="POLY"
  />
</Playground>
