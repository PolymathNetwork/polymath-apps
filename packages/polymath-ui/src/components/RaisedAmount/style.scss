@import '~@polymathnetwork/ui/src/styles/_variables.scss';

.pui-countdown-raised {
  padding: 15px;
  font-size: 36px;
  border-radius: 4px;
  color: $poly-blue;
  background-color: $poly-gray-light;

  > div {
    font-size: 14px;

    &:last-child {
      font-size: 16px;
      font-weight: 300;
      line-height: 25px;
      color: $poly-gray-dark;
      margin-bottom: 0;
    }
  }
}
