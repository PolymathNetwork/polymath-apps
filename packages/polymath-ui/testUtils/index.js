import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
