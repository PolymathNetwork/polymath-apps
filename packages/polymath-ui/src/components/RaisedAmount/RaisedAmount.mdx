---
name: RaisedAmount
---

import { PropsTable } from 'docz';

import Playground from '../DoczPlayground';
import RaisedAmount from './';

# RaisedAmount

<PropsTable of={RaisedAmount} />

## Basic usage

<Playground>
  <RaisedAmount title="Amount Of Funds the STO Will Raise" primaryAmount="10000" tokenAmount="10000" />
</Playground>
