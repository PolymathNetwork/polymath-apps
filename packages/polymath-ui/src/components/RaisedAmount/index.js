import RaisedAmount from './RaisedAmount';

export default RaisedAmount;
