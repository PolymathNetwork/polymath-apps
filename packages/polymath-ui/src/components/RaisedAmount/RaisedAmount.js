import React from 'react';
import styled from 'styled-components';
import Paragraph from '../Paragraph';
import { thousandsDelimiter } from '../../helpers';

import './style.scss';

const Container = styled.div`
  padding: 10px 15px;
  border-radius: 4px;
  background-color: ${({ theme }) => theme.colors.blue[0]};
`;

const LightText = styled.span`
  color: ${({ theme }) => theme.colors.gray[2]};
`;

const RaisedAmount = ({
  title,
  primaryAmount,
  primaryUnit,
  tokenAmount,
  tokenUnit,
}) => (
  <Container>
    <Paragraph as="h4" fontSize={1} mb={3}>
      {title}
    </Paragraph>
    <Paragraph fontSize={5} mb={2}>
      {`${thousandsDelimiter(primaryAmount)} ${primaryUnit}`}
    </Paragraph>
    <LightText>{`${thousandsDelimiter(tokenAmount)} ${tokenUnit}`}</LightText>
  </Container>
);

export default RaisedAmount;
