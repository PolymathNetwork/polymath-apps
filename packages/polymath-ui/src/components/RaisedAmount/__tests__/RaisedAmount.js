import React from 'react';
import { render } from '../../../../testUtils';

import RaisedAmount from '../RaisedAmount';

test('renders without crashing', () => {
  const { container } = render(
    <RaisedAmount title="a title" primaryAmount={555} tokenAmount={111} />
  );

  expect(container.firstChild).toMatchSnapshot();
});
