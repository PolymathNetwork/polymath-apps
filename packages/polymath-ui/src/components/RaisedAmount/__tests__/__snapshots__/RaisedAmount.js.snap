// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="sc-htpNat kOpwYP"
>
  <h4
    class="sc-bwzfXH hEfZSf"
    font-size="1"
  >
    a title
  </h4>
  <div
    class="sc-bwzfXH jJQfzh"
    font-size="5"
  >
    555
  </div>
  <span
    class="sc-bxivhb kXRgtx"
  >
    111
  </span>
</div>
`;
